engine/api/src/test/scala/pl/touk/nussknacker/engine/api/deployment/SimpleProcessStateSpec.scala

```diff
@@ -26,4 +29,38 @@ class SimpleProcessStateSpec extends FunSpec with Matchers with Inside {
     state.status.isFinished shouldBe true
     state.allowedActions shouldBe List(ProcessActionType.Deploy)
   }
+
+  it ("should properly decode StateStatus") {
+    val json = s"""
+      {
+        "clazz": "${SimpleStateStatus.Canceled.getClass.getSimpleName}",
+        "value": "${SimpleStateStatus.Canceled.name}"
+      }
+    """
+
+    val decodedStatus = parser.decode[StateStatus](json) match {
+      case Right(status) => status
+      case Left(error) => throw error
```

Comment:
You shouldn't throw an exception in tests. It will cause test abortion instead of failure. You should use `EitherValues` with `.right.value`

Reply:
Done.